ui/widget.js

```diff
@@ -367,16 +369,33 @@ $.Widget.prototype = {
 		return this;
 	},
 	_setOption: function( key, value ) {
+		var classKey, elements;
+		if ( key === "classes" ) {
+			for ( classKey in value ) {
+				if ( value[ classKey ] !== this.options.classes[ classKey ] ) {
+					if ( this.classObject[ classKey ] && this.classObject[ classKey ].length ) {
+						elements = $( this.classObject[ classKey ].get() );
```

Comment:
the `get()` is because the `_removeClass` call below is going to empty this reference so we need to hold on to it to add the classes back to the same set.